pkgs/by-name/mi/mixxc/package.nix

```diff
@@ -0,0 +1,59 @@
+{
+  lib,
+  rustPlatform,
+  fetchCrate,
+  pkg-config,
+  libpulseaudio,
+  gtk4-layer-shell,
+  gtk4,
+  libxcb,
+  installShellFiles,
+  enableWayland ? false,
+  enableSass ? false,
+  enableX11 ? false,
+}:
+rustPlatform.buildRustPackage rec {
+  pname = "mixxc";
+  version = "0.2.2";
+
+  src = fetchCrate {
+    pname = "mixxc";
+    inherit version;
+    hash = "sha256-Y/9l8t6Vz7yq9T1AyoHnWmIcju1rfcV0S74hiK1fEjo=";
+  };
+
+  cargoHash = "sha256-l9inqqUiLObrqd/8pNobwBbLaiPJD39YK/38CWfDh+Q=";
+
+  cargoBuildFlags = ["--locked"];
+  buildFeatures = with lib;
+    optionals enableWayland ["Wayland"]
+    ++ optionals enableX11 ["X11"]
+    ++ optionals enableSass ["Sass"];
+
+  outputs = [
+    "out"
+    "man"
+  ];
+
+  nativeBuildInputs = [
+    pkg-config
```

Comment:
recommend `wrapGAppsHook4`, just you don't know when people will request it (missing icon theme?)